lib/Service/BoardImportService.php

```diff
@@ -0,0 +1,454 @@
+	/** @var AttachmentMapper */
+	private $attachmentMapper;
+	/** @var ICommentsManager */
+	private $commentsManager;
+	/** @var string */
+	private $system = '';
+	/** @var null|ABoardImportService */
+	private $systemInstance;
+	/** @var array */
+	private $allowedSystems = [];
+	/**
+	 * Data object created from config JSON
+	 *
+	 * @var \stdClass
+	 * @psalm-suppress PropertyNotSetInConstructor
+	 */
+	public $config;
+	/**
+	 * Data object created from JSON of origin system
+	 *
+	 * @var \stdClass
+	 * @psalm-suppress PropertyNotSetInConstructor
+	 */
+	private $data;
+	/**
+	 * @var Board
+	 */
+	private $board;
+
+	public function __construct(
+		IUserManager $userManager,
+		BoardMapper $boardMapper,
+		AclMapper $aclMapper,
+		LabelMapper $labelMapper,
+		StackMapper $stackMapper,
+		AssignmentMapper $assignmentMapper,
+		AttachmentMapper $attachmentMapper,
+		CardMapper $cardMapper,
+		ICommentsManager $commentsManager
+	) {
+		$this->userManager = $userManager;
+		$this->boardMapper = $boardMapper;
+		$this->aclMapper = $aclMapper;
+		$this->labelMapper = $labelMapper;
+		$this->stackMapper = $stackMapper;
+		$this->cardMapper = $cardMapper;
+		$this->assignmentMapper = $assignmentMapper;
+		$this->attachmentMapper = $attachmentMapper;
+		$this->commentsManager = $commentsManager;
+		$this->board = new Board();
+		$this->disableCommentsEvents();
+	}
+
+	private function disableCommentsEvents(): void {
+		if (defined('PHPUNIT_RUN')) {
+			return;
+		}
+		$propertyEventHandlers = new \ReflectionProperty($this->commentsManager, 'eventHandlers');
+		$propertyEventHandlers->setAccessible(true);
+		$propertyEventHandlers->setValue($this->commentsManager, []);
+
+		$propertyEventHandlerClosures = new \ReflectionProperty($this->commentsManager, 'eventHandlerClosures');
+		$propertyEventHandlerClosures->setAccessible(true);
+		$propertyEventHandlerClosures->setValue($this->commentsManager, []);
+	}
+
+	public function import(): void {
+		$this->bootstrap();
+		try {
+			$this->importBoard();
+			$this->importAcl();
+			$this->importLabels();
+			$this->importStacks();
+			$this->importCards();
+			$this->assignCardsToLabels();
+			$this->importComments();
+			$this->importCardAssignments();
+		} catch (\Throwable $th) {
+			throw new BadRequestException($th->getMessage());
+		}
+	}
+
+	public function validateSystem(): void {
+		$allowedSystems = $this->getAllowedImportSystems();
+		$allowedSystems = array_column($allowedSystems, 'internalName');
+		if (!in_array($this->getSystem(), $allowedSystems)) {
+			throw new NotFoundException('Invalid system');
+		}
+	}
+
+	/**
+	 * @param mixed $system
+	 * @return self
+	 */
+	public function setSystem($system): self {
+		$this->system = $system;
+		return $this;
+	}
+
+	public function getSystem(): string {
+		return $this->system;
+	}
+
+	public function getAllowedImportSystems(): array {
+		if (!$this->allowedSystems) {
+			$allowedSystems = glob(__DIR__ . '/BoardImport*Service.php');
```

Comment:
I agree with @juliushaertl, it would be better to import the classes statically at the top of this file and a static list in this method.
I would say it's ok to add the event emission and handling in a future PR.

Reply:
I will dispatch event on this place. What you think @eneiluj and @juliushaertl ?